pandas/core/indexes/interval.py

```diff
@@ -766,8 +766,11 @@ def get_loc(self, key, method=None):
                 key = Interval(left, right, key.closed)
             else:
                 key = self._maybe_cast_slice_bound(key, 'left', None)
-
-            start, stop = self._find_non_overlapping_monotonic_bounds(key)
+            try:
+                start, stop = self._find_non_overlapping_monotonic_bounds(key)
+            except TypeError:
+                # get_loc should raise KeyError
```

Comment:
Can you add here a comment as Tom proposed (`TODO(py3): use raise from.`) ?

Reply:
you can now use raise from (as we are PY3 only)